deployment/upgrade_iambic_version_for_generic_git_provider_lambda/upgrade_lambda.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import os
+import time
+
+import boto3
+
+REGION_NAME = os.environ.get("AWS_REGION", "us-east-1")
+IAMBIC_CODE_BUILD_PROJECT_NAME = os.environ.get(
+    "IAMBIC_CODE_BUILD_PROJECT_NAME", "iambic_code_build"
+)
+IAMBIC_FUNCTION_NAME = os.environ.get(
+    "IAMBIC_FUNCTION_NAME", "iambic_generic_git_provider_webhook"
+)
+IAMBIC_REPOSITORY_NAME = os.environ.get(
+    "IAMBIC_REPOSITORY_NAME", "iambic-ecr-public/iambic/iambic"
+)
+IAMBIC_CF_LAMBDA_STACK_NAME = os.environ.get(
+    "IAMBIC_CF_LAMBDA_STACK_NAME", "IAMbicGenericGitProviderLambda"
+)
+IAMBIC_TARGET_VERSION = os.environ.get("IAMBIC_TARGET_VERSION", "latest")
+
+
+def start_code_build_with_pin_version(ver):
+    code_build_client = boto3.client("codebuild", region_name=REGION_NAME)
+
+    response = code_build_client.start_build(
+        projectName=IAMBIC_CODE_BUILD_PROJECT_NAME,
+        environmentVariablesOverride=[
+            {
+                "name": "IMAGE_TAG",
+                "value": ver,
+                "type": "PLAINTEXT",
+            },
+        ],
+    )
+
+    build_id = response["build"]["id"]
+    # FIXME explain why this is stalling for builds
```

Comment:
FIXME okay to leave?

Reply:
context: this is a cicd script sitting in their build system. so I attempt to not import anything that requires another pip install. 

I switch this to standard logging module in python standard library.